lib/inline-imports.js

```diff
@@ -86,7 +86,48 @@ function inlineImports(fileContents, options = {}, visited = new Set()) {
   return result.join('\n');
 }
 
+function getFilename(filePath, options = {}) {
+  const { resolveImport, resolveOptions = {} } = options;
+
+  const { basedir } = resolveOptions;
+  if (typeof basedir !== 'string') {
+    throw new Error('inlineImports requires options.resolverOptions.basedir be set');
+  }
+  const resolve = typeof resolveImport === 'function' ? resolveImport : require('resolve').sync;
+  return resolve(filePath, resolveOptions);
+}
+
+function getFileContents(filePath, options = {}) {
+  const { fs } = options;
+  const resolvedFs = typeof fs === 'function' ? fs : require('fs');
+  return resolvedFs.readFileSync(filePath, 'utf8');
+}
+
+/*
+ * Inline any import statements of the graphql document from `fullFilePath`
+ *
+ * @typedef {{ basedir: string, ...args: [any]}} ResolverOptions
+ * @property {string} basedir - The base directory to use when resolving import identifiers.  For most platforms this will be the absolute path to the directory containing the graphql query file with imports.
+ * @property {...args} [any] - any other properties on a `ResolverOptions` will be passed through to the resolver.
+ * @param {string} fullFilePath - The graphql document filepath to inline imports into
+ * @param {object} options
+ * @param {(identifier: string, options: ResolverOptions) => string} [options.resolveImport=require.resolve] - a function for resolving imports.  It must support relative paths as well as any package specifiers that may be present as identifiers in the import statements of `fileContents`.  For invalid import identifiers, it should throw an error. Defaults to `resolve.sync` (i.e. treating imported packages as node modules).
+ * @param {ResolverOptions} options.resolveOptions - options that are passed to `options.resolveImport` when resolving imports.
+ * @param {boolean} [throwIfImportNotFound=true] - whether or not to throw for invalid imports, or to silently ignore them.
+ */
+function inlineImportsFromFile(fullFilePath, options = {}) {
```

Comment:
My inclination would be to add an option `filePath` to `inlineImports`.

`inlineImportsFromFile` would then just resolve `fullFilePath` and delegate to `inlineImports` rather than duplicate the body of `inlineImports` and also needing to explicitly prepopulate the visited set.


Is there a reason to prefer the current approach?